### PR TITLE
[scanlog] Add EndIndex option

### PR DESCRIPTION
**Pull request by pavelkalinnikov**
The option allows scanning arbitrary `[StartIndex, EndIndex)` subranges of the log.

### CODE REVIEW

File: scanner/scanner.go [hunk 1]
```diff
@@ -51,9 +51,13 @@ type ScannerOptions struct { // nolint:golint
 	// Number of fetched entries to buffer on their way to the callbacks
 	BufferSize int
 
-	// Log entry index to start fetching & matching at
+	// Log entry index to start fetching & matching at.
 	StartIndex int64
 
+	// Log entry index to finish fetching & matching at (non-inclusive).
+	// Scan up to the log's end if EndIndex == -1.
+	EndIndex int64
```
Comment (daviddrysdale) [inline scanner/scanner.go]: What happens to all of the `scanner` users who implicitly leave this at the default value of `EndIndex: 0` ?

Comment (pavelkalinnikov) [reply scanner/scanner.go]: Didn't realize those. How about making `EndIndex==0` a safe default meaning "end of log" instead of -1?

**Comment by pavelkalinnikov** [comment]
Merged #198 in. PTAL.

File: scanner/scanlog/scanlog.go [hunk 2]
```diff
@@ -54,6 +54,7 @@ var (
 	numWorkers    = flag.Int("num_workers", 2, "Number of concurrent matchers")
 	parallelFetch = flag.Int("parallel_fetch", 2, "Number of concurrent GetEntries fetches")
 	startIndex    = flag.Int64("start_index", 0, "Log index to start scanning at")
+	endIndex      = flag.Int64("end_index", 0, "Log index to end scanning at (non-inclusive, 0 - end of Log)")
```
Comment (daviddrysdale) [inline scanner/scanlog/scanlog.go]: maybe `0 for end of log` or `0 = end of Log` ?

Comment (pavelkalinnikov) [reply scanner/scanlog/scanlog.go]: Done.

**Pull request merged**